src/main/java/org/codehaus/mojo/versions/SetPropertyMojo.java

```diff
@@ -73,24 +85,54 @@
     protected void update( ModifiedPomXMLEventReader pom )
         throws MojoExecutionException, MojoFailureException, XMLStreamException
     {
-        Property propertyConfig = new Property( property );
-        propertyConfig.setVersion( newVersion );
+        Property[] propertiesConfig = null;
+        String properties = "";
+        if (!StringUtils.isEmpty(propertiesVersionsFile) ) {
+            logWrongConfigWarning();
+            getLog().debug( "Reading properties and versions to update from file: " + propertiesVersionsFile );
+            PropertiesVersionsFileReader reader = new PropertiesVersionsFileReader(propertiesVersionsFile);
+            try {
+                reader.read();
+            } catch (FileNotFoundException e) {
+                getLog().error("Error while reading property file " + propertiesVersionsFile, e);
```

Comment:
Just from looking at the code: shouldn't the plugin fail if the properties file is not readable? I guess, to continue without being able to read the file, the result of the operation is unclear.

Reply:
Oh, right! Fixed by throwing a MojoFailureException